src/App.tsx

```diff
@@ -1,25 +1,44 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './components/API/api';
 
 const App: React.FC = () => {
   const [
     selectedUserId,
-    // setSelectedUserId,
+    setSelectedUserId,
   ] = useState(0);
 
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([]);
+
+  async function todosGetter() {
+    const result = await getTodos();
+
+    setCurrentTodos(result);
+  }
+
+  useEffect(() => {
+    todosGetter();
+  }, [selectedUserId]);
```

Comment:
there is no need in this dependence, you will do requests to todos on every peek of userId

Reply:
There is need. Todos getter updates state. So it's infinite loop without it

Reply:
You don't need to load todos every time when selectedUserId change
You should load them only once, when your component mounted on the page (use componentDidMount method)
If you use hook, so you should pass an empty array as a second argument to make this functionality the same as for componentDidMount method